server/pbenchinacan/run-pbench-in-a-can

```diff
@@ -1,24 +1,125 @@
 export PB_SERVER_IMAGE_NAME=${PB_SERVER_IMAGE_NAME:-"pbench-server-ci"}
 export PB_SERVER_IMAGE_TAG=${PB_SERVER_IMAGE_TAG:-"$(cat jenkins/branch.name)"}
 export PB_POD_NAME=${PB_POD_NAME:-"pbench-in-a-can_${PB_SERVER_IMAGE_TAG}"}
 
-export PB_COMMON_IMAGE_PULL_POLICY=${PB_COMMON_IMAGE_PULL_POLICY:-"Always"}
+export PB_COMMON_IMAGE_PULL_POLICY=${PB_COMMON_IMAGE_PULL_POLICY:-"always"}
 export PB_ELASTICSEARCH_IMAGE_PULL_POLICY="${PB_ELASTICSEARCH_IMAGE_PULL_POLICY:-${PB_COMMON_IMAGE_PULL_POLICY}}"
 export PB_POSTGRESQL_IMAGE_PULL_POLICY="${PB_POSTGRESQL_IMAGE_PULL_POLICY:-${PB_COMMON_IMAGE_PULL_POLICY}}"
 export PB_KEYCLOAK_IMAGE_PULL_POLICY="${PB_KEYCLOAK_IMAGE_PULL_POLICY:-${PB_COMMON_IMAGE_PULL_POLICY}}"
 export PB_SERVER_IMAGE_PULL_POLICY="${PB_SERVER_IMAGE_PULL_POLICY:-${PB_COMMON_IMAGE_PULL_POLICY}}"
 
 export PB_DASHBOARD_DIR="${PB_DASHBOARD_DIR:-${PWD}/dashboard/build/}"
 
-envsubst < $(dirname ${0})/pod.yml | podman play kube -
+podman pod create --name ${PB_POD_NAME} --hostname pbenchinacan -p 9200:9200 -p 5432:5432 -p 8090:8090 -p 8080:8080
+
+podman run \
+  --rm \
+  --name "${PB_POD_NAME}-elasticsearch" \
+  --pod ${PB_POD_NAME} \
+  --workdir /usr/share/elasticsearch \
+  --pull ${PB_ELASTICSEARCH_IMAGE_PULL_POLICY} \
+  --umask 0000 \
+  --restart no \
+  --pids-limit 0 \
+  -e PATH=/usr/share/elasticsearch/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin \
+  -e HOME=/root \
+  -e TERM=xterm \
+  -e container=oci \
+  -e discovery.type=single-node \
+  -e ES_JAVA_OPTS="-Xms1g -Xmx1g"\
+  -e xpack.security.enabled=false \
+  -e ES_PORT=9200 \
+  -d \
+  ${PB_CONTAINER_IMAGE_PREFIX}/${PB_ELASTICSEARCH_IMAGE_NAME}:${PB_ELASTICSEARCH_IMAGE_TAG} \
+  /bin/tini -- /usr/local/bin/docker-entrypoint.sh eswrapper
+
+podman run \
+  --rm \
+  --pod ${PB_POD_NAME} \
+  --name "${PB_POD_NAME}-postgres" \
+  --workdir /opt/app-root/src \
+  --pull ${PB_POSTGRESQL_IMAGE_PULL_POLICY} \
+  --umask 0000 \
+  --user 26:26 \
+  --restart no \
+  --pids-limit 0 \
+  --entrypoint container-entrypoint \
+  -e POSTGRESQL_USER=pbenchcontainer \
+  -e POSTGRESQL_PASSWORD=pbench \
+  -e POSTGRESQL_DATABASE=pbenchcontainer \
+  -e HOME=/var/lib/pgsql \
+  -e CONTAINER_SCRIPTS_PATH=/usr/share/container-scripts/postgresql \
+  -e APP_ROOT=/opt/app-root \
+  -e POSTGRESQL_VERSION="13" \
+  -e PGUSER=postgres \
+  -e STI_SCRIPTS_URL=image:///usr/libexec/s2i \
+  -d \
+  ${PB_CONTAINER_IMAGE_PREFIX}/${PB_POSTGRESQL_IMAGE_NAME}:${PB_POSTGRESQL_IMAGE_TAG} \
+  run-postgresql -p 5432
+
+podman run \
+  --rm \
+  --pod ${PB_POD_NAME} \
+  --name "${PB_POD_NAME}-keycloak" \
+  --pull ${PB_KEYCLOAK_IMAGE_PULL_POLICY} \
+  --umask 0000 \
+  --restart no \
+  --pids-limit 0 \
+  -e KEYCLOAK_ADMIN=admin \
+  -e KEYCLOAK_ADMIN_PASSWORD=admin \
+  -d \
+  ${PB_CONTAINER_IMAGE_PREFIX}/${PB_KEYCLOAK_IMAGE_NAME}:${PB_KEYCLOAK_IMAGE_TAG} \
+  start-dev --health-enabled=true --http-port=8090
+
+source server/pbenchinacan/load_keycloak.sh
```

Comment:
The `source` doesn't hurt, but `load_keycloak.sh` is set up as an independent bash script with shebang and all, and doesn't seem to need to hand local shell context back up to the caller. Why is it sourced???

Reply:
It is sourced because @npalaska has chosen to execute it in the context of the caller instead of in a subprocess...I don't have a problem with that.

Yeah, it _could_ be executed "as a command" (I think that the "x bit" is set on the file...), but I don't see that it **has** to be.

We expect `load_keycloak.sh` to be used in the deployment of Staging and Production servers.  If we expect it to be executed as a command there (as opposed to being `source`'d or `bash`'d), then, yeah, we should probably do the same here.  (But, I don't think it's worth _three_ question marks!!! 😉)

Reply:
Yes, we can, but we don't need to, and it looks odd. Among other things, it made me search the code to see where it was expecting to return shell context to the caller.